pkg/kapis/devops/v1alpha3/handler.go

```diff
@@ -46,23 +46,24 @@ func newDevOpsHandler(devopsClient devopsClient.Interface, k8sClient k8s.Client)
 	}
 }
 
-// devopsproject handler about get/list/post/put/delete
+// GetDevOpsProject handler about get/list/post/put/delete
 func (h *devopsHandler) GetDevOpsProject(request *restful.Request, response *restful.Response) {
 	workspace := request.PathParameter("workspace")
-	devops := request.PathParameter("devops")
+	devopsProject := request.PathParameter("devops")
+	generateNameFlag := request.QueryParameter("generateName")
```

Comment:
Not necessary. We can use `xx == 'true'`

Reply:
I'd prefer to use strconv.ParseBool to parse this flag. It's more flexible and reduces the use of constants. Please see also: 
https://github.com/kubesphere/ks-devops/blob/f1c3c105c78e21a7a2d21fe11186f97b2b909ebb/pkg/server/params/params.go#L123-L129.

You can consider use this utility instead.

Reply:
I still don't have a reason to parse it to a boolean type. Using it as a string is enough.

Reply:
Because this query parameter is an exact boolean type instead of string. We should keep same query parameter style with other similar APIs as well.

BTW, don't forget to add a parameter description at https://github.com/kubesphere/ks-devops/blob/56aeda5423c825a41027792503d0cdee2bfe4ceb/pkg/kapis/devops/v1alpha3/register.go#L178

Reply:
> this query parameter is an exact boolean type instead of string

No, it is not. It only looks like a boolean type. For this API, `generateName` could be `"true"` or other cases.

Reply:
> BTW, don't forget to add a parameter description at

How about adding it in other PRs? It is not part of this PR.

Reply:
> No, it is not. It only looks like a boolean type. For this API, `generateName` could be `"true"` or other cases.

I'm sorry I cannot image what other cases should be like. Could you show me some cases?

> How about adding it in other PRs? It is not part of this PR.

IMO, the query parameter description is exactly related with this PR. So that we could understand how to use the `generateName` query parameter.

Reply:
So, what is the problem if I use it like this:

`if generateName == "true"` this way is simple, and no extra cost (cast one type to another one)

and, what are the pros if turn it to boolean type?

Reply:
API consumers are able to pass `"1", "t", "T", "true", "TRUE", "True"` to a boolean query parameter instead of only `true`. We should obey one rule of API definition, as I said before: we should keep same query parameter style with other similar APIs as well instead of following our mind.

```go
// ParseBool returns the boolean value represented by the string.
// It accepts 1, t, T, TRUE, true, True, 0, f, F, FALSE, false, False.
// Any other value returns an error.
func ParseBool(str string) (bool, error) {
	switch str {
	case "1", "t", "T", "true", "TRUE", "True":
		return true, nil
	case "0", "f", "F", "false", "FALSE", "False":
		return false, nil
	}
	return false, syntaxError("ParseBool", str)
}
```

Reply:
> BTW, don't forget to add a parameter description at

I didn't understand your meaning. Updated the description.

Reply:
For this case, it only supports `true` instead of other cases. If you insist, I will give up using `true` or `false`. I don't understand why we really need to discuss so much about this.

Reply:
By the way, I don't want to waste my time discussing this anymore. It does not help with anything.